### DISCUSSION TITLE
10 analog sensor, 74HC4067 adc mulplexer with esp32

### DISCUSSION THREAD

**Discussion by 10david11** [arendst/Tasmota#15764] [Q&A] [answered]
My new project is making an esp32 tasmota weather station with more analog sensor than esp32 can handle. I ordered a 74HC4067 adc mulplexer from aliexpress.

Please help me:
How to wire the 74HC4067 adc mulplexer with esp32?
How to setup on the web ui?
How to configure the 16 analog ports?

**Comment by sfromis** [comment]
If you just want an easy solution, get another ESP32 on the side, much easier to deal with. Cheap too.

That analog multiplexer currently has no Tasmota support, but "someone" with the urge could create a driver. As it appears to be running via 4 address select pins, it should be relatively simple even without that, maybe doable with rules or remote commands.

**Reply by barbudor** [reply]
Or get I2C ADCs such as ADS1115
- 16 bits resolution
- Easier to make a proper analog circuit when the ADC is not in the MPU
- Up to 4 devices supported by the driver so total 16 ADC inputs

With such solution, Tamsota see the 16 ADC as independant channels
With a multiplexer, Tasmota will not be aware and you'll get confused by `ANALOGn` having different meaning

**Comment by pkkrusty** [comment, marked as answer]
Gemu2015 made a nice script for that module, in the scripting section.

**Reply by 10david11** [reply]
I found it and added the script to Tasmota 11.1.0, but it was not success. I dont know how to wire the modul with esp32 and what I have to do on the ui.

**Reply by sfromis** [reply]
In case others are looking, I assume that the script is this one:
https://tasmota.github.io/docs/Scripting-Language/#multiplexing-a-single-adc-with-cd4067-breakout

**Reply by gemu2015** [reply]
in scripts sometimes the pins are defined in the script itself, not in the tasmota pin UI
check the script code, here pins 12-15 are used

; define output pins for multiplexer
spinm(12 O)
spinm(13 O)
spinm(14 O)
spinm(15 O)

**Reply by gemu2015** [reply]
in any case you have to deal with scripting to adapt the code to your needs.
this example was meant for soil humidity sensors for flowers

**Reply by pkkrusty** [reply]
@gemu2015 In scripting documentation, you wrote `adc(fac (pin)) = get adc value (on ESP32 can select pin) fac is number of averaged samples (power of 2: 0..7)` suggesting that `adc(4 32)` would get 16 samples. But in the multiplexer example, the comment says 
```
; get adc value into array, average 4 values
; this is for ESP32 here on pin 32
mux[mcnt+1]=adc(4 32)
```
but according to docs, 4 samples should be `adc(2 32)`. Can you clarify which is correct?

**Reply by gemu2015** [reply]
the first parameter is the power of 2 factor for averaging.
1 = 2
2 = 4
3 = 8
...

so adc(2 32) gets 4 samples.

in the multiplexer example it should actually be
mux[mcnt+1]=adc(2 32) for 4 samples
